### PR TITLE
Script Generator: add save to disk functionality

### PR DESCRIPTION
**Pull request by simonfernandes**
### Description of work

A user should now be able to save a script as a .py file to a disk location of their choice. The script written to file should match that of the script preview content.
### To test

https://github.com/ISISComputingGroup/IBEX/issues/1564
### Acceptance criteria

_List the acceptance criteria for the PR_

---
#### Code Review
- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/GUI-Coding-Conventions)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit tests in place? Are the unit tests small and test the a class in isolation?
- [ ] Are there [system tests](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/System-Testing-with-RCPTT) in place? Do they test a minimal set of functionality and leave the gui as close as possible to its original state?
- [ ] Did any existing system test break as a result of the current changes? 
- [x] Have the changes been documented in the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev). If so, do they describe the changes appropriately?
- [ ] Has the [manual system tests spreadsheet](https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Manual-system-tests) been updated?
### Functional Tests
- [ ] Do changes function as described? Add comments below that describe the tests performed.
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has developer documentation been updated if required?
### Final Steps
- [ ] Reviewer has moved the [release notes](https://github.com/ISISComputingGroup/IBEX/wiki/ReleaseNotes_Dev) entry for this ticket in the "Changes merged into master" section

### CODE REVIEW

File: base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/View.java [hunk 1]
```diff
@@ -174,9 +179,38 @@ public void widgetSelected(SelectionEvent e) {
...
+        		
+        		if (filename != null) {
+        			String script = builder.getScript();
+        			PrintWriter writer;
+					try {
+						writer = new PrintWriter(filename);
+						writer.print(script);
+						writer.close();
+					} catch (FileNotFoundException e1) {
+						e1.printStackTrace();
```
Comment (John-Holt-Tessella) [inline base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/View.java]: User should be told if file can not be saved

File: base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/View.java [hunk 2]
```diff
@@ -174,9 +179,38 @@ public void widgetSelected(SelectionEvent e) {
 		btnWrite.addSelectionListener(new SelectionAdapter() {
             @Override
             public void widgetSelected(SelectionEvent e) {
-                script = builder.getScript();
+                String defaultFileName = ".py";
```
Comment (John-Holt-Tessella) [inline base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/View.java]: .py is a magic value put at top and refernced (in several places)

File: base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/View.java [hunk 3]
```diff
@@ -174,9 +179,38 @@ public void widgetSelected(SelectionEvent e) {
...
+        		dialog.setFilterExtensions(new String[] {"*.py"});
+        		dialog.setFilterPath("c:\\");
+        		dialog.setFileName(defaultFileName);
+
+        		String filename = dialog.open();
+        		
+        		if (filename != null) {
+        			String script = builder.getScript();
+        			PrintWriter writer;
+					try {
```
Comment (John-Holt-Tessella) [inline base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/View.java]: unindent

File: base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/View.java [hunk 4]
```diff
@@ -174,9 +179,38 @@ public void widgetSelected(SelectionEvent e) {
...
             public void widgetSelected(SelectionEvent e) {
-                script = builder.getScript();
+                String defaultFileName = ".py";
+
+        		FileDialog dialog = new FileDialog(saveShell, SWT.SAVE);
+
+        		dialog.setFilterNames(new String[] {"Python File (*.py)", 
+        				"All Files (*.*)" });
+        		dialog.setFilterExtensions(new String[] {"*.py"});
+        		dialog.setFilterPath("c:\\");
```
Comment (John-Holt-Tessella) [inline base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/View.java]: default to c:\scripts please

File: base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/View.java (hunk 3 above)
Comment (mattclarke) [inline base/uk.ac.stfc.isis.ibex.ui.scriptgenerator/src/uk/ac/stfc/isis/ibex/ui/scriptgenerator/View.java]: Is this a github thing? It looks fine in eclipse

**Comment by DominicOram** [comment]
As per the original ticket can you also add the functionality to save as csv?

**Pull request merged**